src/game/client/components/chat.cpp

```diff
@@ -800,12 +827,50 @@ void CChat::OnRender()
 			m_InputUpdate = false;
 		}
 
-		TextRender()->TextEx(&Cursor, m_Input.GetString()+m_ChatStringOffset, m_Input.GetCursorOffset()-m_ChatStringOffset);
-		static float MarkerOffset = TextRender()->TextWidth(0, 8.0f, "|", -1, -1.0f)/3;
-		CTextCursor Marker = Cursor;
-		Marker.m_X -= MarkerOffset;
-		TextRender()->TextEx(&Marker, "|", -1);
-		TextRender()->TextEx(&Cursor, m_Input.GetString()+m_Input.GetCursorOffset(), -1);
+		//render buffered text
+		if(m_Mode == CHAT_NONE) {
+			//truncate text to 29 chars
+			char aText[30];
+			str_copy(aText, m_Input.GetString(), sizeof(aText));
+
+			//add dots when string excesses length
+			if(m_Input.GetLength() >= (int)(sizeof(aText))-2)
+			{
+				for(int i = 0; i < 3; ++i)
+				{
+					aText[sizeof(aText)-2-i] = '.';
+				}
+			}
+			aText[29] = 0;
+
+			TextRender()->TextColor(1.0f, 1.0f, 1.0f, 0.5f);
+			TextRender()->TextEx(&Cursor, aText, -1);
+
+			//render helper annotation
+			CTextCursor InfoCursor;
+			TextRender()->SetCursor(&InfoCursor, 2.0f, y+12.0f, CategoryFontSize*0.75, TEXTFLAG_RENDER);
+
+			//find keyname and format text
+			char aKeyName[64];
+			m_pClient->m_pBinds->GetKey(GetCommandName(m_ChatBufferMode), aKeyName, sizeof(aKeyName));
+
+			char aInfoText[128];
+			str_format(aInfoText, sizeof(aInfoText), "Press %s to resume chatting", aKeyName);
+
+			TextRender()->TextColor(1, 1, 1, Blend);
```

Comment:
Since Blend is a parameter, i will change line 846 and remove line 860

Reply:
Huh, this code change is not equivalent to the previous code.

There used to be 
```TextRender()->TextColor(1.0f, 1.0f, 1.0f, 0.5f);
TextRender()->TextColor(1.0f, 1.0f, 1.0f, Blend);```

now there is
```
TextRender()->TextColor(1.0f, 1.0f, 1.0f, Blend);```

Should 0.5f really be used instead of Blend?